packages/yourloops/lib/selected-team/selected-team.hook.tsx

```diff
@@ -36,39 +36,48 @@ export function useSelectedTeamProviderCustomHook(): SelectedTeamContextResult {
   const { getMedicalTeams, getPrivateTeam } = useTeam()
   const medicalTeams = getMedicalTeams()
   const privateTeam = getPrivateTeam()
+  const availableTeams = useMemo(() => ([...medicalTeams, privateTeam]), [medicalTeams, privateTeam])
 
-  const getDefaultTeam = (): Team => {
+  const getDefaultTeam = useCallback((): Team => {
     if (!medicalTeams.length) {
       return privateTeam
     }
 
     return TeamUtils.sortTeamsByName(medicalTeams)[0]
-  }
+  }, [medicalTeams, privateTeam])
 
-  const getDefaultTeamId = (): string => {
-    const availableTeams = [...medicalTeams, privateTeam]
+  const getTeamToSelect = useCallback((): Team => {
     const localStorageTeamId = localStorage.getItem(LOCAL_STORAGE_SELECTED_TEAM_ID_KEY)
     const isValidTeamId = availableTeams.some((team: Team) => team.id === localStorageTeamId)
 
     if (!isValidTeamId) {
       const defaultTeam = getDefaultTeam()
-      const defaultId = defaultTeam.id
+      localStorage.setItem(LOCAL_STORAGE_SELECTED_TEAM_ID_KEY, defaultTeam.id)
 
-      localStorage.setItem(LOCAL_STORAGE_SELECTED_TEAM_ID_KEY, defaultId)
-      return defaultId
+      return defaultTeam
     }
-    return localStorageTeamId
-  }
 
-  const [selectedTeamId, setSelectedTeamId] = useState<string>(() => getDefaultTeamId())
+    return availableTeams.find((team: Team) => team.id === localStorageTeamId)
+  }, [availableTeams, getDefaultTeam])
+
+  const [selectedTeam, setSelectedTeam] = useState<Team>(() => getTeamToSelect())
 
   const selectTeam = (teamId: string): void => {
-    setSelectedTeamId(teamId)
+    const team = availableTeams.find((team: Team) => team.id === teamId)
+    if (!team) {
+      return
+    }
+    setSelectedTeam(team)
     localStorage.setItem(LOCAL_STORAGE_SELECTED_TEAM_ID_KEY, teamId)
   }
 
+  useEffect(() => {
+    const team = getTeamToSelect()
+    setSelectedTeam(team)
+  }, [getDefaultTeam, getTeamToSelect, medicalTeams])
```

Comment:
<!-- Nitpicks are trivial preference-based requests. These should be non-blocking by nature.  -->
**🤓 nitpick:** The dependency to `medicalTeams` is redundant (as `getDefaultTeam` is already dependant on `getDefaultTeam`). These lint rules make it unlikely that a useEffect/useMemo/useCallback has to depend from something that is not used by himself (often, the dependency list contains dependencies that end up depending on what you want)‏
```suggestion
  }, [getDefaultTeam, getTeamToSelect])
```